### PR TITLE
Used single account for both account and contract

### PR DESCRIPTION
**Pull request by dk-neon**

### CODE REVIEW

**Review by afalaleev — DISMISSED** [review]

**Review by afalaleev — DISMISSED** [review]

File: proxy/common_neon/neon_instruction.py [hunk 1]
```diff
@@ -31,7 +32,8 @@ class EvmInstruction(Enum):
     HolderCreate = b'\x24'  # 36
     HolderDelete = b'\x25'  # 37
     HolderWrite = b'\x26'  # 38
-
+    DepositV03 = b'\x27'
```
Comment (s-medvedev) [inline proxy/common_neon/neon_instruction.py]: Please, add comment with decimal representation

File: proxy/common_neon/solana_interactor.py [hunk 2]
```diff
@@ -304,7 +282,7 @@ def get_block_slot(self, commitment='confirmed') -> int:
         }
         return self._send_rpc_request('getSlot', opts)['result']
 
-    def get_account_info(self, pubkey: PublicKey, length=256, commitment='processed') -> Optional[AccountInfo]:
+    def get_account_info(self, pubkey: PublicKey, length=64 * 1024, commitment='processed') -> Optional[AccountInfo]:
```
Comment (s-medvedev) [inline proxy/common_neon/solana_interactor.py]: Specify unlimitted size by default (`length=0`) like in `get_neon_code_info` was.

File: proxy/common_neon/solana_interactor.py [hunk 3]
```diff
@@ -317,7 +295,7 @@ def get_account_info(self, pubkey: PublicKey, length=256, commitment='processed'
             }
 
         result = self._send_rpc_request('getAccountInfo', str(pubkey), opts)
-        # self.debug(f"{json.dumps(result, sort_keys=True)}")
+        self.debug(f"{json.dumps(result, sort_keys=True)}")
```
Comment (s-medvedev) [inline proxy/common_neon/solana_interactor.py]: Comment back this string (this debug data can lead to dramatically increase proxy logs)

File: proxy/common_neon/data.py [hunk 4]
```diff
@@ -21,6 +22,7 @@ def __init__(self):
         self._evm_step_cnt = 0
         self._alt_list: List[str] = []
         self._account_dict: NeonAccountDict = {}
+        self._additional_resize_steps = False
```
Comment (s-medvedev) [inline proxy/common_neon/data.py]: Pass required resize steps as number and increase calculated iteration count in iterative send strategy.

File: proxy/indexer/neon_ix_decoder.py [hunk 5]
```diff
@@ -374,10 +353,15 @@ def execute(self) -> bool:
...
+    _name = 'Deposit3'
+    _ix_code = 0x27
+    _is_deprecated = False
+
+
 def get_neon_ix_decoder_list() -> List[Type[DummyIxDecoder]]:
     return [
-        CreateAccount2IxDecoder,
-        ResizeStorageAccountIxDecoder,
+        CreateAccount3IxDecoder,
```
Comment (s-medvedev) [inline proxy/indexer/neon_ix_decoder.py:364]: For relaunch history it is ok to forget how to decode deprecated instructions.
But for upgrade scenario we should decode ALL historical instructions.

File: proxy/common_neon/neon_instruction.py (hunk 1 above)
Comment (dk-neon) [reply proxy/common_neon/neon_instruction.py]: Done.

File: proxy/common_neon/solana_interactor.py (hunk 3 above)
Comment (dk-neon) [reply proxy/common_neon/solana_interactor.py]: Done.

File: proxy/common_neon/data.py (hunk 4 above)
Comment (dk-neon) [reply proxy/common_neon/data.py]: Done.

File: proxy/common_neon/solana_interactor.py (hunk 2 above)
Comment (dk-neon) [reply proxy/common_neon/solana_interactor.py]: Done. I'm using length=None by default. It gives us possibility to query zero-sized account's data as well (in order for check existence, by example).

File: proxy/indexer/neon_ix_decoder.py (hunk 5 above)
Comment (dk-neon) [reply proxy/indexer/neon_ix_decoder.py:364]: Done in #956.

File: proxy/common_neon/data.py [hunk 6]
```diff
@@ -34,13 +36,25 @@ def evm_step_cnt(self) -> int:
...
         self._account_dict = account_dict
         self._evm_step_cnt = evm_step_cnt
+        self._additional_resize_steps = NeonTxExecCfg.resolve_additional_resize_steps(account_dict)
         return self
 
+    @staticmethod
+    def resolve_additional_resize_steps(emulated_result: NeonEmulatedResult) -> int:
+        for account in emulated_result["accounts"]:
+            if int(account["additional_resize_steps"] or 0):
+                return True
```
Comment (s-medvedev) [inline proxy/common_neon/data.py]: Why do you return `True/False`?

Comment (dk-neon) [reply proxy/common_neon/data.py]: Oops. Fixed. :)

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**